Add error checking to ctrs in clickNewContract

filippsen opened the pull request:
Relates to issue #16

javier-tarazaga commented on src/components/projecteditor/control.js:
```diff
@@ -697,9 +697,11 @@ export default class DevkitProjectEditorControl extends Component {
-                const contract=ctrs[ctrs.length-2];
-                const item=contract.props.state.children[0];
-                if(this.props.router.panes) this.props.router.panes.openItem(item);
+                if(ctrs) {
```
I would check also that the length is >=2 otherwise would not make sense and a new exeception will be raised. Why ctrs[ctrs.lenght-2] though?

filippsen commented:
@javier-tarazaga `ctrs` is to reference all the contracts listed in a project. As for "why length minus 2", it is due to the following:
1. `ctrs[ctrs.length]` is `undefined`. The last valid element is of index `ctrs.length-1`;
2. The last valid element is not a contract, but the "make contract" prop.

That results in the last added contract being of index `ctrs.length-2`. In practice, as it is today, provided `ctrs` is valid, it will simply be of length `2` or more. I added the check you mentioned, regardless of that assumption (commit 380b3465)

Your questioning made me revisit the whole purpose of the `clickNewContract` function. The last time I focused on getting rid of the "uncaught error", but the actual cause is due to the fact that index `0` has been changed through the course of development to contain the "Transaction Logs" feature. The correct starting index for retrieving "contracts children" is `1`. 

The purpose of that code is to bring up the contract "Configure" menu when creating a new file. The one that reads "Edit Contract <Contract Name Here>. After you click "New file", it is expected to bring up that panel. I added commit 7a1f4eaa to fix that.

Thanks!

javier-tarazaga commented:
Thanks @filippsen! I am back and will take a look tomorrow during the day to make sure we can merge this PR.

javier-tarazaga commented on src/components/projecteditor/control.js:
```diff
@@ -696,10 +696,12 @@ export default class DevkitProjectEditorControl extends Component {
-                const item=contract.props.state.children[0];
-                if(this.props.router.panes) this.props.router.panes.openItem(item);
+                const ctrs=projectItem.props.state.children[1].props.state._children;
+                if(ctrs && ctrs.length >= 2) {
```
@filippsen could you add the comments here of why choosing .children[1] and ctrs.lenghth >=2 as it nos not obvious in any sense for the reader of this code? With a small comment in the code there wont be any more questions of why those values.
